### ISSUE TITLE
Running on Apple Silicon

### ISSUE THREAD

**Issue by chiaqf** [saharmor/dalle-playground#30]

**Comment by danielchalef** [comment]
JAX is not yet fully supported on the Apple Silicon GPU. See here: https://github.com/google/jax/issues/8074#issuecomment-1148982985

You may get it work -- very slowly -- using the CPU version of GPU.

**Comment by Birch-san** [comment]
running dalle-playground on-CPU on M1 Max is 60x slower than the [huggingface demo](https://huggingface.co/spaces/dalle-mini/dalle-mini).

i.e. it can generate 1 image in 27 minutes. 117% CPU usage.
you can get more throughput by submitting concurrent tasks. best throughput I achieved was with 5 concurrent tasks;
5 1-image tasks in 52 mins. 550% CPU usage. so 10.4 mins per pic.

there is a path to get it running on-GPU via IREE/Vulkan-over-MoltenVK. first we need to get IREE backend working:  
https://github.com/google/iree/issues/9361  
work in-progress:  
https://github.com/google/iree/pull/9378

**Comment by haaakon** [comment]
are you running mega on M1 Max?

**Comment by haaakon** [comment]
how did you get past   this error when running app.py? 
```
  from vqgan_jax.modeling_flax_vqgan import VQModel
ModuleNotFoundError: No module named 'vqgan_jax'
```

**Comment by Birch-san** [comment]
Yes, I'm using M1 Max. It's only concurrent in the sense that I open multiple browser tabs of the dalle-playground frontend, and ask for 1 image via each browser tab simultaneously. It's still an order of magnitude longer duration and less throughput than the online dalle-mini demo. I've asked and apparently that online demo uses the dalle-mega model anyway.

**Comment by haaakon** [comment]
Ah was hoping to use a bigger model to have some improved resolution on the images generated, didn't know the online version was also using mega
